src/main/java/springweb/courseproject/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+import springweb.courseproject.repository.order.OrderRepository;
+import springweb.courseproject.repository.shoppingcart.ShoppingCartRepository;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderResponseDto createOrder(Long userId, CreateOrderRequestDto createOrderRequestDto) {
+        ShoppingCart cart = shoppingCartRepository.findByUserId(userId).orElseThrow(
+                () -> new EntityNotFoundException("Shopping cart with user id "
+                        + userId + " not found")
+        );
+        if (cart.getCartItems().isEmpty()) {
+            throw new DataProcessingException("You can't place an order with empty shopping cart");
+        }
+        Order order = new Order();
+        order.setUser(cart.getUser());
+        order.setStatus(Status.PENDING);
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(createOrderRequestDto.getShippingAddress());
+        Set<OrderItem> orderItemSet = cart.getCartItems().stream()
+                .map(cartItem -> {
+                    OrderItem orderItem = orderItemMapper.toOrderItem(cartItem);
+                    orderItem.setOrder(order);
+                    return orderItem;
+                })
+                .collect(Collectors.toSet());
+        order.setOrderItems(orderItemSet);
+        BigDecimal total = cart.getCartItems().stream()
+                .map(cartItem -> cartItem.getBook().getPrice().multiply(
+                        BigDecimal.valueOf(cartItem.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+        order.setTotal(total);
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderResponseDto> getAllUsersOrders(Pageable pageable, Long userId) {
+        return orderRepository.findAllByUserId(pageable, userId).stream()
+                .map(orderMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public OrderResponseDto updateOrder(Long orderId, UpdateOrderRequestDto updateOrderRequestDto) {
+        Order order = orderRepository.findById(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Order with order id " + orderId + " not found")
+        );
+        order.setStatus(updateOrderRequestDto.getStatus());
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderItemResponseDto> getOrderItemsById(Long orderId, Long userId) {
+        Order order = orderRepository.findById(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Order with order id " + orderId + " not found")
+        );
+        checkIfOrderBelongsToUser(order, userId);

```

Comment:
You can check on request level 
```suggestion
        Order order = orderRepository.findByIdAndUserId(orderId, userId).orElseThrow(
                () -> new EntityNotFoundException("Order with order id " + orderId + " not found...")
        );
```